src/Symfony/Component/Filesystem/Tests/FilesystemTest.php

```diff
@@ -370,6 +370,65 @@ public function testFilesExists()
         $this->assertTrue($this->filesystem->exists($basePath.'folder'));
     }
 
+    public function testFilesExistsAndOfTypeRegular()
+    {
+        $basePath = $this->workspace.\DIRECTORY_SEPARATOR.'directory'.\DIRECTORY_SEPARATOR;
+
+        mkdir($basePath);
+        touch($basePath.'file1');
+
+        $this->assertTrue($this->filesystem->exists($basePath.'file1', null));
+        $this->assertTrue($this->filesystem->exists($basePath.'file1', \Symfony\Component\Filesystem\Filesystem::FILE_TYPE_REGULAR));
+        $this->assertFalse($this->filesystem->exists($basePath.'file1', \Symfony\Component\Filesystem\Filesystem::FILE_TYPE_DIRECTORY));
+    }
+
+    public function testFilesExistsAndOfTypeDirectory()
+    {
+        $basePath = $this->workspace.\DIRECTORY_SEPARATOR.'directory'.\DIRECTORY_SEPARATOR;
+
+        mkdir($basePath);
+        mkdir($basePath.'folder');
+
+        $this->assertTrue($this->filesystem->exists($basePath.'folder', null));
+        $this->assertFalse($this->filesystem->exists($basePath.'folder', \Symfony\Component\Filesystem\Filesystem::FILE_TYPE_REGULAR));
+        $this->assertTrue($this->filesystem->exists($basePath.'folder', \Symfony\Component\Filesystem\Filesystem::FILE_TYPE_DIRECTORY));
+    }
+
+    public function testFilesExistsAndFullSetIsOfTheSameType()
+    {
+        $basePath = $this->workspace.\DIRECTORY_SEPARATOR.'directory'.\DIRECTORY_SEPARATOR;
+
+        mkdir($basePath);
+        touch($basePath.'file');
+        touch($basePath.'file2');
+        mkdir($basePath.'folder');
+        mkdir($basePath.'folder2');
+
+        $filesOnly = new \ArrayObject([
+            $basePath.'file', $basePath.'file2',
+        ]);
+
+        $dirsOnly = new \ArrayObject([
+            $basePath.'folder', $basePath.'folder2',
+        ]);
+
+        $mixedFilesDirs = new \ArrayObject([
+            $basePath.'file', $basePath.'folder', $basePath.'file2', $basePath.'folder2',
```

Comment:
if you make it a multi-line array, you should put 1 item per line. Otherwise, there is no point making it a multi-line array

Reply:
Sure, I just copy-pasted the next test case which had the set on one line, and did it the same way. I can def make it multi-line though.